client/src/main/java/io/pravega/client/control/impl/Controller.java

```diff
@@ -364,6 +365,14 @@
      */
     CompletableFuture<Transaction.Status> checkTransactionStatus(final Stream stream, final UUID txId);
 
+    /**
+     * Gets an {@link AsyncIterator} on TxnResponse in stream.
+     *
+     * @param stream The name of the stream for which to list completed transaction.
+     * @return An {@link AsyncIterator} which can be used to iterate over all TxnResponse in the stream.
```

Comment:
Api have continuationToken as a params in request. Its not always start iterating from beginning, api is in paginated manner and result of api is dependent on value of continuationToken.

Reply:
You misunderstand my question. I meant it is iterating over "completed transactions" IE ones that have been committed previously. Prior to this PR any committed transaction could be forgotten about and all data related to it cleaned up. This appears to introduce a requirement to retain this information, but it's not clear how long it will be retained. For example is it expected that all transactions on a stream since its creation will always be included? That is surprising and perhaps undesirable because it could be millions or billions of transactions. 

The exact semantics of this need to be made clear.

Reply:
**listCompletedTransactions** api is fetching completed transaction from completedTransactionsBatch-0 table only. Due to a bug **Commit Transaction batch number always shows as 0**  [ref](https://github.com/pravega/pravega/issues/6603). Lifetime of completedTransactionsBatch- table is still remain same as it was earlier. completedTransactionsBatch- tables will get deleted periodically. This api will return only most recent completed transactions.

Reply:
Let's fill in a little more details about what is and is not included and specifically if there is a cap. (There should be some limit)
(Also I'm not clear on the connection to the bug, and can't we just fix that?)

Reply:
**listCompletedTransactions** will provides us list of completed transaction from most recent batch by reading the entries from **completedTransactionsBatch-** table. Removed continuationToken from api because  **completedTransactionsBatch-** can have maximum 200 entries, so can fetch all the records in a single call instead of fetching it in a paginated way. This api will not impact lifetime of completedTransactionsBatch- table. 
I will start working on above bug. It is already in my pipeline. It will not impact api much. Just need to replace table name from completedTransactionsBatch-0 with completedTransactionsBatch-0latestBatchNo.